crates/ark/src/lsp/completions/completion_item.rs

```diff
@@ -375,34 +374,44 @@ pub(super) unsafe fn completion_item_from_namespace(
     parameter_hints: &ParameterHints,
 ) -> anyhow::Result<CompletionItem> {
     // First, look in the namespace itself.
-    if let Some(item) = completion_item_from_symbol(
+    match completion_item_from_symbol(
         name,
         namespace,
         Some(package),
         PromiseStrategy::Force,
         parameter_hints,
     ) {
-        return item;
+        Ok(item) => return Ok(item),
+        Err(_) => {
+            // The only error we anticipate is the case where `namespace`
+            // doesn't have a binding for `name`, because the associated object
+            // has been imported and re-exported. For example, the way dplyr
+            // imports and re-exports `rlang::.data` or `tidyselect::all_of()`.
+            // In such a case, we'll succeed below, when we try again in the
+            // imports environment.
+        },
     }
 
     // Otherwise, try the imports environment.
     let imports = ENCLOS(namespace);
-    if let Some(item) = completion_item_from_symbol(
+    match completion_item_from_symbol(
         name,
         imports,
         Some(package),
         PromiseStrategy::Force,
         parameter_hints,
     ) {
-        return item;
+        Ok(item) => return Ok(item),
+        Err(err) => {
+            // This is really unexpected.
+            bail!(
```

Comment:
Here and below: We no longer use `bail!` and instead do the explicit `return Err(anyhow!(...))`. This way the `return` stands out.